### PR TITLE
ACEseqQC

### PR DESCRIPTION
**Pull request by vinjana**
Sorry a big merge with some changes affecting just change code. Tests are not complete yet, but please have a look to gain some understanding what I changed and how it might interact with your previous changes.

### CODE REVIEW

File: RoddyCore/src/de/dkfz/roddy/config/ToolFileGroupParameter.groovy [hunk 1]
```diff
@@ -67,8 +68,12 @@ class ToolFileGroupParameter extends ToolEntry.ToolParameter<ToolFileGroupParame
     }
 
     @Override
-    ToolFileGroupParameter clone() {
-        return new ToolFileGroupParameter(scriptParameterName, groupClass, genericFileClass,  passOptions, indexOptions, files?.collect { ToolFileParameter tfp -> tfp.clone() })
+    public ToolFileGroupParameter clone() {
+        List<ToolFileParameter> _files = new LinkedList<ToolFileParameter>()
```
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/config/ToolFileGroupParameter.groovy]: It is not a list of files, it is a list of file parameters.
Why did you move away from the collect method?

File: RoddyCore/test/de/dkfz/roddy/config/converters/YAMLConverterTest.groovy [hunk 2]
```diff
@@ -19,7 +19,7 @@ public class YAMLConverterTest {
         String yaml = """
             configuration:
             availableAnalyses: {analysis: [{_id: onScriptTest, _configuration: testAnalysis, _useplugin: TestPluginWithJarFile}, {_id: WGS, _configuration: bisulfiteCoreAnalysis, _useplugin: AlignmentAndQCWorkflows}]}
-            configurationvalues: {cvalue: [{_name: inputBaseDirectory, _value: $USERHOME/temp/roddyLocalTest/testproject/vbp, _type: path}, {_name: outputBaseDirectory, _value: $USERHOME/temp/roddyLocalTest/testproject/rpp, _type: path}, {_name: INDEX_PREFIX, _value: '${indexPrefix_bwa06_methylCtools_mm10_GRC}', _type: path}, {_name: CLIP_INDEX, _value: $DIR_EXECUTION/analysisTools/qcPipeline/trimmomatic/adapters/tagmentationPlassGroup.fa, _type: path}]}
+            configurationvalues: {cvalue: [{_name: inputBaseDirectory, _value: \$USERHOME/temp/roddyLocalTest/testproject/vbp, _type: path}, {_name: outputBaseDirectory, _value: $USERHOME/temp/roddyLocalTest/testproject/rpp, _type: path}, {_name: INDEX_PREFIX, _value: '${indexPrefix_bwa06_methylCtools_mm10_GRC}', _type: path}, {_name: CLIP_INDEX, _value: $DIR_EXECUTION/analysisTools/qcPipeline/trimmomatic/adapters/tagmentationPlassGroup.fa, _type: path}]}
```
Comment (dankwart-de) [inline RoddyCore/test/de/dkfz/roddy/config/converters/YAMLConverterTest.groovy]: Not every $ is \ed

**Comment by vinjana** [comment]
The two issues you mentioned are reverted to the version in the development branch. One was a merge mistake, in the other I tried to fix another test, but as the test is actually not really implemented I did not proceed further. Is the YAMLConverter functional?

**Comment by dankwart-de** [comment]
Looks good to me for the moment.
No the YAMLConverter is not functional and also not used! We can move the YAML stuff to a custom feature branch but currently I don't see that we will enable YAML within the next weeks.
